MTGToolbox.Repository/CardRepository.cs

```diff
@@ -1,52 +1,32 @@
-﻿using System;
+﻿using Microsoft.EntityFrameworkCore;
+using MTGToolbox.Abstract;
 using System.Collections.Generic;
 using System.Linq;
-using System.Threading.Tasks;
-using MTGToolbox.Core;
 
 namespace MTGToolbox.Repository
 {
-    public class CardRepository : ICardRepository, IDisposable
+    public class CardRepository : ICardRepository
     {
         private MTGToolboxContext context;
-        private bool disposed = false;
 
         public CardRepository(MTGToolboxContext context)
         {
             this.context = context;
         }
 
-        public IEnumerable<Card> GetCards()
+        public IEnumerable<ICard> GetCards()
         {
             return context.Cards.ToList();
         }
 
-        public Card GetCardById(int id)
+        public ICard GetCardById(int id)
         {
             return context.Cards.Find(id);
         }
 
-        public Card GetCardByName(string name)
+        public ICard GetCardByName(string name)
         {
-            return context.Cards.Find(name);
-        }
-
-        protected virtual void Dispose(bool disposing)
-        {
-            if (!this.disposed)
-            {
-                if (disposing)
-                {
-                    context.Dispose();
-                }
-            }
-            this.disposed = true;
-        }
-
-        public void Dispose()
-        {
-            Dispose(true);
-            GC.SuppressFinalize(this);
+            return context.Cards.SingleOrDefault(c => EF.Functions.Like(c.Name, name));
```

Comment:
You were using `Find(…)` to look by name but I think that only works by ID. So for string matching, you can do something like this. Alternatively you can also execute a stored proc.